scripts/Diffraction/isis_powder/routines/focus.py

```diff
@@ -56,32 +56,18 @@ def _focus_one_ws(
     perform_vanadium_norm,
     absorb,
     sample_details,
-    vanadium_path,
+    vanadium_ws,
     empty_can_subtraction_method,
     paalman_pings_events_per_point=None,
 ):
     run_details = instrument._get_run_details(run_number_string=run_number)
-    if perform_vanadium_norm:
-        _test_splined_vanadium_exists(instrument, run_details)
 
-    # Subtract empty instrument runs, as long as this run isn't an empty, user hasn't turned empty subtraction off, or
-    # The user has not supplied a sample empty
-    is_run_empty = common.runs_overlap(run_number, run_details.empty_inst_runs)
-    summed_empty = None
-    if not is_run_empty and instrument.should_subtract_empty_inst() and not run_details.sample_empty:
-        if os.path.isfile(run_details.summed_empty_inst_file_path):
-            logger.warning("Pre-summed empty instrument workspace found at " + run_details.summed_empty_inst_file_path)
-            summed_empty = mantid.LoadNexus(Filename=run_details.summed_empty_inst_file_path)
-        else:
-            summed_empty = common.generate_summed_runs(empty_ws_string=run_details.empty_inst_runs, instrument=instrument)
-    elif run_details.sample_empty:
-        scale_factor = 1.0
-        if empty_can_subtraction_method != "PaalmanPings":
-            scale_factor = instrument._inst_settings.sample_empty_scale
-        # Subtract a sample empty if specified ie empty can
-        summed_empty = common.generate_summed_runs(
-            empty_ws_string=run_details.sample_empty, instrument=instrument, scale_factor=scale_factor
-        )
+    summed_empty = _load_summed_empty(
```

Comment:
I've added a separate comment\question about whether the logic relating to the summed empty needs to be different for OSIRIS. Let's discuss that later on.
If the conclusion is that the logic doesn't need to be different I would prefer it if we could leave this section of code as it was ie not pull L67-84 out into a separate function. You can argue this makes the code look better but it makes it harder to see whether anything has changed. Also there's another PR that rearranges this piece of code (see https://github.com/mantidproject/mantid/pull/35280) so don't want to end up with a merge conflict.

Reply:
This is reverted back